ports/wch/Makefile

```diff
@@ -0,0 +1,125 @@
+# Include the core environment definitions; this will set $(TOP).
+include ../../py/mkenv.mk
+
+# qstr definitions (must come before including py.mk)
+QSTR_DEFS = qstrdefsport.h
+
+# MicroPython feature configurations
+MICROPY_ROM_TEXT_COMPRESSION ?= 1
+
+# Include py core make definitions.
+include $(TOP)/py/py.mk
+
+# Path you your toolchain and openocd installation, leave empty if already in system PATH
+TOOLCHAIN_ROOT = /opt/wch/mounriver-studio-toolchain-riscv/bin/
+OPENOCD_ROOT   = /opt/wch/mounriver-studio-toolchain-openocd/bin/
+
+CROSS_COMPILE ?= $(TOOLCHAIN_ROOT)riscv-none-embed-
```

Comment:
Please consider an early startup test that looks if ${TOOLCHAIN_ROOT} is a directory and ${CROSS_COMPILE}gcc is a file. Issue a helpful error message so the developer knows where to look in the doc and/or configuration files.

Sometimes hardcoded paths just can't be avoided, but it's worth calling out expectations so another user can meet them.